lib/engine/src/trap/frame_info.rs

```diff
@@ -42,6 +42,14 @@ pub struct GlobalFrameInfo {
     ranges: BTreeMap<usize, ModuleInfoFrameInfo>,
 }
 
+/// Returns whether the `pc`, according to globally registered information,
+/// is a wasm trap or not.
+pub fn is_wasm_pc(pc: usize) -> bool {
+    let frame_info = FRAME_INFO.read().expect("can't acquire the FrameInfo lock");
```

Comment:
When could this happen anyways? Would it be enough to have the same code running in two threads crashing at the same time, triggering the `is_wasm_pc` in both threads?

Reply:
No, this is https://doc.rust-lang.org/std/sync/type.LockResult.html, the reason we just `unwrap` these is because it's not recoverable in general. I think the `expect` message is a bit misleading here as it sounds like it's doing a non-blocking acquire on the lock and failing, but it's really that the data covered by the lock cannot be guaranteed to be in a valid state anymore.

Panicking while holding a lock poisons the lock (because the data is in an undefined state), the best thing to do > 99% of the time is to propagate the panic by just `unwrap`ing it because the system is likely in an unrecoverable state, at least as far as that data is concerned.

Reply:
Cool, I'll move it back into `unwrap`